python/lsst/pipe/tasks/make_direct_warp.py

```diff
@@ -409,30 +459,48 @@ def run(self, inputs, sky_info, visit_summary):
             )
             warpedExposure.setPsf(psfWarped)
 
+            if final_warp.photoCalib is not None and warpedExposure.photoCalib is not None:
+                ratio = (
+                    final_warp.photoCalib.getInstFluxAtZeroMagnitude()
+                    / warpedExposure.photoCalib.getInstFluxAtZeroMagnitude()
+                )
+            else:
+                ratio = 1
```

Comment:
Do we always expect the first logic branch here to fire in practice?  I'm wondering if we can somehow make it an error rather than a silent change in behavior if a `PhotoCalib` goes incorrectly missing upstream.

Reply:
I primarily intended this to run without issues if one were to give simulated images without a `photoCalib`. Would it be better to have a trivial `photoCalib` that is the same in all exposures instead?

Reply:
On real data, for the very first `calexp`, the else branch would fire since `final_warp` has no `photoCalib` attached to it. The first `calexp`'s `photoCalib` gets attached to the `final_warp` and others are scaled accordingly. In the past, the ordering was not consistent resulting in apparent differences. Lauren had patched it up by ordering the refs by detector ID, but I take that butler gives the inputs in the same order now.

Reply:
Yes, I think expecting simulated data to have a trivial `PhotoCalib` is reasonable.

So I think we just need to raise if the `calexp` `PhotoCalib` is absent, while allowing the warp's to be absent (for the first iteration).

Reply:
Alright, branching only based on the warp's `photoCalib`. I don't want to impose that this exception is made only for the first iteration. If there are no good pixels from the first visit, then the `photoCalib` comes from the second visit.